library/nmcli.py

```diff
@@ -0,0 +1,735 @@
+        # if self.type == 'ethernet':
+        #     cmd.append = 'con'
+        #     cmd.append = self.name
+        #     cmd.append('con-name')
+        #     if self.cname is not None:
+        #         cmd.append(self.cname)
+        #     elif self.cname is None:
+        #         if self.ifname is not None:
+        #             cmd.append(self.ifname)
+        #     cmd.append('ifname')
+        #     cmd.append(self.ifname)
+        #     cmd.append('type')
+        #     cmd.append(self.type)
+        #     if self.ip4 is not None:
+        #         cmd.append('ip4')
+        #         cmd.append(self.ip4)
+        #     if self.gw4 is not None:
+        #         cmd.append('gw4')
+        #         cmd.append(self.gw4)
+        #     if self.dns4 is not None:
+        #         cmd.append('dns4')
+        #         cmd.append(self.dns4)
+        #     if self.ip6 is not None:
+        #         cmd.append('ip6')
+        #         cmd.append(self.ip6)
+        #     if self.gw6 is not None:
+        #         cmd.append('gw6')
+        #         cmd.append(self.gw4)
+        #     if self.dns6 is not None:
+        #         cmd.append('dns6')
+        #         cmd.append(self.dns6)
+        return cmd
+
+    def create_connection_bridge(self):
+        cmd=[self.module.get_bin_path('nmcli', True)]
+        # format for creating bridge interface
+        return cmd
+
+    def modify_connection_bridge(self):
+        cmd=[self.module.get_bin_path('nmcli', True)]
+        # format for modifying bridge interface
+        return cmd
+
+    def create_connection_vlan(self):
+        cmd=[self.module.get_bin_path('nmcli', True)]
+        # format for creating ethernet interface
+        return cmd
+
+    def modify_connection_vlan(self):
+        cmd=[self.module.get_bin_path('nmcli', True)]
+        # format for modifying ethernet interface
+        return cmd
+
+    def create_connection(self):
+        cmd=[]
+        if self.type=='team':
+            cmd=self.create_connection_team()
+        elif self.type=='team-slave':
+            cmd=self.create_connection_team_slave()
+        elif self.type=='bond':
+            cmd=self.create_connection_bond()
+        elif self.type=='bond-slave':
+            cmd=self.create_connection_bond_slave()
+        elif self.type=='ethernet':
+            cmd=self.create_connection_ethernet()
+        elif self.type=='bridge':
+            cmd=self.create_connection_bridge()
+        elif self.type=='vlan':
+            cmd=self.create_connection_vlan()
+        return self.execute_command(cmd)
+
+    def remove_connection(self):
+        # self.connection_down()
+
+        cmd=[self.module.get_bin_path('nmcli', True)]
+        cmd.append('con')
+        cmd.append('del')
+        cmd.append(self.cname)
+        return self.execute_command(cmd)
+
+    def modify_connection(self):
+        cmd=[]
+        if self.type=='team':
```

Comment:
you can create a dict of type to method call

commands = { 'team': self.modify_connection_team, 'team-slave', ... }
return self.execute_command(commands[self.type])